monkey/monkey_island/cc/services/attack/attack_config.py

```diff
@@ -0,0 +1,168 @@
+                    continue
+                try:
+                    should_remove = not AttackConfig.should_enable_field(array_field['attack_techniques'], attack_techniques)
+                except KeyError:
+                    # Monkey schema field contains not yet implemented technique
+                    continue
+                # If exploiter's attack technique is disabled, disable the exploiter/scanner/PBA
+                AttackConfig.r_alter_array(monkey_config, key, array_field['enum'][0], remove=should_remove)
+
+    @staticmethod
+    def set_booleans(attack_techniques, monkey_config, monkey_schema):
+        """
+        Sets boolean type fields, like "should use mimikatz?" in monkey's config according to ATT&CK matrix
+        :param attack_techniques: ATT&CK techniques dict. Format: {'T1110': True, ...}
+        :param monkey_config: Monkey island's configuration
+        :param monkey_schema: Monkey configuration schema
+        """
+        for key, value in monkey_schema['properties'].items():
+            AttackConfig.r_set_booleans([key], value, attack_techniques, monkey_config)
+
+    @staticmethod
+    def r_set_booleans(path, value, attack_techniques, monkey_config):
+        """
+        Recursively walks trough monkey configuration (DFS) to find which boolean fields needs to be set and sets them
+        according to ATT&CK matrix.
+        :param path: Property names that leads to current value. E.g. ['monkey', 'system_info', 'should_use_mimikatz']
+        :param value: Value of config property
+        :param attack_techniques: ATT&CK techniques dict. Format: {'T1110': True, ...}
+        :param monkey_config: Monkey island's configuration
+        """
+        if isinstance(value, dict):
+            dictionary = {}
+            # If 'value' is a boolean value that should be set:
+            if 'type' in value and value['type'] == 'boolean' and 'attack_techniques' in value:
+                try:
+                    AttackConfig.set_bool_conf_val(path,
+                                                   AttackConfig.should_enable_field(value['attack_techniques'],
+                                                                                    attack_techniques),
+                                                   monkey_config)
+                except KeyError:
+                    # Monkey schema has a technique that is not yet implemented
+                    pass
+            # If 'value' is dict, we go over each of it's fields to search for booleans
+            elif 'properties' in value:
+                dictionary = value['properties']
+            else:
+                dictionary = value
+            for key, item in dictionary.items():
+                path.append(key)
+                AttackConfig.r_set_booleans(path, item, attack_techniques, monkey_config)
+        # Method enumerated everything in current path, goes back a level.
+        del path[-1]
```

Comment:
move this line from here to after the recursive call to r_set_booleans.

It makes much more sense that whoever appended the path, removes the appended part. Makes it much more readable